File: two-sums/two-sums.cpp
#include <iostream>
#include <vector>

//I hate that I have to add this line
using namespace std;

//Leet Ratings
//       Speed  Memory
//Total  675ms  10.2MB    
//Beats  6.87% 73.65%
//
//This is so slow because it, among other things, does not do a reverse search and one of the test cases
//has a ton of numbers where the correct indexes are at the end.



vector<int> twoSum(vector<int>& nums, int target) {
        
        int first = 0;
        int second = 0;
        bool done = false;
        int length = nums.size();
        for(int i = 0 ; i < length ;  ++i){
            if(done == true){
              break;
            }
            for (int x = 0; x < length ; ++x){


                if( nums[x] + nums[i] == target & x != i){
                    done = true;
                    first = x;
                    second = i;
                    
                    break;
                }


            }



        }

        vector<int> vals = {first , second};
        return (vals);
    }


int main(){


    vector<int> sums = {3, 2, 4};    
    twoSum(sums, 6);
    cout << twoSum(sums, 6)[0] << twoSum(sums, 6) [1] << endl;

}

File: two-sums/two-sums.py

"""
Leet Ratings
       Speed  Memory
Total  1005ms   15MB    
Beats  38.22% 67.55%

"""

def twoSum( nums: list[int], target: int):
    count1 = len(nums) - 1
    count2 = len(nums) - 1
    while count1 >= 0:
        count2 = len(nums) - 1
        while count2 >= 0:
            if nums[count1] + nums[count2] == target and count1 != count2:
                return [count1, count2]
            count2 -= 1
        count1 -= 1
    return [0,0]


nums = [0 , 1 , 4, 5 ,7 , 67]


print(twoSum(nums , 67))
